### ISSUE TITLE
zsh: illegal hardware instruction interpreter

### ISSUE THREAD

**Issue by datasauruscom** [KillianLucas/open-interpreter#279]
### Describe the bug

I set up my local conda env and installation process is smooth.
But I keep having the issue like code llama setup is failed, as following,

surui@M1 ~ % interpreter --local

Open Interpreter will use Code Llama for local execution. Use your arrow keys to set up the model.                                                                                                          

[?] Parameter count (smaller is faster, larger is more capable): 7B
 > 7B
   13B
   34B

[?] Quality (lower is faster, higher is more capable): Medium | Size: 4.24 GB, RAM usage: 6.74 GB
   Low | Size: 3.01 GB, RAM usage: 5.51 GB
 > Medium | Size: 4.24 GB, RAM usage: 6.74 GB
   High | Size: 7.16 GB, RAM usage: 9.66 GB

[?] Use GPU? (Large models might crash on GPU, but will run more quickly) (Y/n): Y

[?] This instance of `Code-Llama` was not found. Would you like to download it? (Y/n): Y

▌ Failed to install Code-LLama.                                                                                                                                                                           

We have likely not built the proper Code-Llama support for your system.                                                                                                                                     

( Running language models locally is a difficult task! If you have insight into the best way to implement this across platforms/architectures, please join the Open Interpreter community Discord and       
consider contributing the project's development. )                                                                                                                                                          

Please press enter to switch to GPT-4 (recommended).                                                                                                                                                        

### Reproduce

conda create -n codeinterpreter python=3.11.4 -y
conda activate codeinterpreter
pip install open-interpreter 
interpreter --local

### Expected behavior

it should be able to install codellama quickly as most of the tutorial shows. 

### Open Interpreter version

 0.1.3

### Python version

3.11.4

### Operating System name and version

macOS 13.5.2

**Comment by hoeoek** [comment]
I had a similar problem. For me, I installed the llama python bindings manually using pip and then it worked as expected.

**Comment by jordanbtucker** [comment]
Please see https://github.com/KillianLucas/open-interpreter/blob/main/docs/MACOS.md and report back if that doesn't solve the issue.

**Comment by datasauruscom** [comment]
Thanks worked for me now.

**Comment by datasauruscom** [comment]
Model found at /Users/ou/Library/Application Support/Open Interpreter/models/codellama-7b-instruct.Q2_K.gguf                                                                                              
zsh: illegal hardware instruction  interpreter 

It began to download but I still have the above probelm and still can not install open interpreter.

**Comment by sonykey2003** [comment]
> 

doesn't work for my M1 MacBookPro

Model found at /Users/ssong/Library/Application Support/Open                                          
Interpreter/models/codellama-13b-instruct.Q4_K_M.gguf                                                 
[1]    70504 illegal hardware instruction  interpreter

**Comment by derekbar90** [comment]
Same issue here

```
interpreter --local

Open Interpreter will use Code Llama for local execution. Use your arrow keys to set up the model.

[?] Parameter count (smaller is faster, larger is more capable): 13B
   7B
 > 13B
   34B

[?] Quality (lower is faster, higher is more capable): Medium | Size: 8.06 GB, RAM usage: 10.56 GB
   Low | Size: 5.66 GB, RAM usage: 8.16 GB
 > Medium | Size: 8.06 GB, RAM usage: 10.56 GB
   High | Size: 13.83 GB, RAM usage: 16.33 GB

[?] Use GPU? (Large models might crash on GPU, but will run more quickly) (Y/n): y

[1]    31504 illegal hardware instruction  interpreter --local
```

**Comment by jordanbtucker** [comment]
I think there is something wrong with that model file. Other users have reported it doesn't work on their macs. It's the only one that seems to be a problem.

**Comment by derekbar90** [comment]
This was due to improper versions of Python and miniconda. I used open interpreter to auto-fix these issues so that the llama-cpp-python was installed properly and outside of open interpreter. 

TLDR: The error is due to the wrong llama-cpp-python being installed. You can use a OpenAI key to have open-interpreter verify your python and llama-cpp-python install

**Comment by jordanbtucker** [comment]
> The error is due to the wrong llama-cpp-python being installed. You can use a OpenAI key to have open-interpreter verify your python and llama-cpp-python install

Thanks. Can you provide more detailed steps for other users?

**Comment by Notnaton** [comment]
Old, uses ooba which is deprecated
